docs/_source/docs/hooks.rst

```diff
@@ -44,23 +44,72 @@ Available Hooks
 cmd
 ~~~
 
-Executes the argument string in the shell as a Python subprocess.
-
-For more information about how this works, see the `subprocess documentation`_
+Executes a command through the shell.
 
 Syntax:
 
 .. code-block:: yaml
 
+   # Default shell.
    <hook_point>:
      - !cmd <shell_command>
 
-Example:
+   # Another shell.
+   <hook_point>:
+     - !cmd
+         command: <shell_command>
+         shell: <shell_name_or_path>
+
+Pass the command string as the only argument to use the default shell.
+
+On POSIX the default shell is ``sh``. On Windows it's usually ``cmd.exe``, but ``%ComSpec%`` may
+override that.
+
+Format the command string exactly as you would type it at the shell prompt. This includes
+quotes and backslashes to escape filenames with spaces in them.
+
+.. code-block:: yaml
+
+   hooks:
+     before_update:
+       - !cmd "echo 'Hello, world!'"
+
+This example succeeds. Its standard output messages appear between status messages from Sceptre.
+
+.. code-block::
+
+   [2023-09-03 01:06:28] - test - Launching Stack
+   [2023-09-03 01:06:29] - test - Stack is in the CREATE_COMPLETE state
+   Hello, world!
+   [2023-09-03 01:06:31] - test - Updating Stack
+   [2023-09-03 01:06:31] - test - No updates to perform.
+
+Pass named arguments ``args`` and ``executable`` to use a different shell.
+
+Format the executable name exactly as you would type it to start the shell. For example, if Bash is
+in system path, you can write ``bash``; otherwise, you need to write the absolute path such as
+``/bin/bash``.
 
 .. code-block:: yaml
 
-   before_create:
-     - !cmd "echo hello"
+   hooks:
+     before_update:
+       - !cmd
+           command: "if foo; then baz; else qux; fi"
+           shell: "bash"
+
+This example fails because none of the ``foo``, ``baz``, and ``quz`` commands exist. Its standard error messages sit between Sceptre's status messages and a Python traceback.
```

Comment:
seems odd to me to use an example that fails.  i'm not sure I understand what this example is trying to demonstrate specifically?  why not just do  `command: "echo $0"` which should echo "bash"?

Reply:
Yes, good point. I will split this into two examples to make it clearer.

* one that echoes the shell name and uses Bash syntax to prove we are using the Bash shell
* one that fails (could be any shell) to show what Sceptre does with the shell error

Reply:
@zaro0508 I used your simpler example and added a new one to show the error behavior. I added a PowerShell example too since it's a common choice on Windows. How does it look now?

Reply:
looks good